### PR TITLE
Don't allow empty or long labels

### PR DESCRIPTION
**Pull request by krukow**

### CODE REVIEW

**Comment by kamstrup** [comment]
Looks good to me. Main concern, as with the similar PR on appium, is that 128 chars is too restrictive.

Also just thinking - maybe we should add some thorough parameter validation in the Event constructor as well? - Now that we have our grubby hands in here anyway :-D

**Comment by krukow** [comment]
@kamstrup please re-review

File: src/main/java/com/microsoft/appcenter/event/Event.java [hunk 1]
```diff
@@ -2,7 +2,18 @@
...
     private Event(String id, EventType type, String testName, String className, String parameters) {
+        if (null == id || id.length() < 1) {
+            throw new IllegalArgumentException("id must be a non-empty string");
+        }
+        if (null == testName || testName.length() < 1) {
+            throw new IllegalArgumentException("testName must be a non-empty string");
+        }
+        if (null == className || className.length() < 1) {
+            throw new IllegalArgumentException("className must be a non-empty string");
+        }
```
Comment (kamstrup) [inline src/main/java/com/microsoft/appcenter/event/Event.java:16]: I assume parameters is free form and allowed to be null.

File: src/main/java/com/microsoft/appcenter/event/Event.java [hunk 2]
```diff
@@ -12,6 +23,9 @@ private Event(String id, EventType type, String testName, String className, Stri
     }
 
     public static Event createLabel(String id, String testName, String className, String run, String label) {
+        if (null == label || label.length() < 1 || label.length() > MAX_LABEL_SIZE) {
+            throw new IllegalArgumentException("labels must be non-empty strings of length <= " + MAX_LABEL_SIZE);
+        }
         Event event = new Event(id, EventType.label, testName, className, run);
```
Comment (kamstrup) [inline src/main/java/com/microsoft/appcenter/event/Event.java:29]: As noted on the appium pr I would rather be truncating at 160 chars, but if you prefer a hard limit of 256 I am fine with that also. There is a technical reason why << 256 is preferable. On some filesystems paths are limited to 256 chars, and we see labels and other identifiers ending up in paths from time to time... But no biggie, just a note.

**Review by kamstrup — APPROVED** [review]

**Pull request merged**